tests/unit/test_business_logic.py

```diff
@@ -1441,6 +1442,201 @@ def test_add_file_to_request_with_filetype(bll, filetype, success):
             bll.add_file_to_request(release_request, path, author, filetype=filetype)
 
 
+def test_update_file_in_request_invalid_file_type(bll):
+    author = factories.create_user("author", ["workspace"], False)
+
+    relpath = UrlPath("path/file.foo")
+    workspace = factories.create_workspace("workspace")
+    factories.write_workspace_file(workspace, relpath)
+    release_request = factories.create_release_request(
+        "workspace",
+        user=author,
+    )
+
+    # This code reproduces bll.add_file_to_request() - except without the
+    # safeguards - in order that we can give update_file_in_request()
+    # the correct context to hit this test
+
+    workspace = bll.get_workspace(release_request.workspace, author)
+    src = workspace.abspath(relpath)
+    file_id = store_file(release_request, src)
+    manifest = workspace.get_manifest_for_file(relpath)
+    group_name = "default"
+
+    audit = AuditEvent.from_request(
+        request=release_request,
+        type=AuditEventType.REQUEST_FILE_ADD,
+        user=author,
+        path=relpath,
+        group=group_name,
+        filetype=RequestFileType.OUTPUT.name,
+    )
+
+    bll._dal.add_file_to_request(
+        request_id=release_request.id,
+        group_name=group_name,
+        relpath=relpath,
+        file_id=file_id,
+        filetype=RequestFileType.OUTPUT,
+        timestamp=manifest["timestamp"],
+        commit=manifest["commit"],
+        repo=manifest["repo"],
+        size=manifest["size"],
+        job_id=manifest["job_id"],
+        row_count=manifest["row_count"],
+        col_count=manifest["col_count"],
+        audit=audit,
+    )
+    with pytest.raises(
+        bll.RequestPermissionDenied, match=r"Cannot update file of type"
+    ):
+        bll.update_file_in_request(release_request, relpath, author)
+
+
+def test_update_file_in_request_not_updated(bll):
+    author = factories.create_user("author", ["workspace"], False)
+
+    relpath = UrlPath("path/file.txt")
+    workspace = factories.create_workspace("workspace")
+    factories.write_workspace_file(workspace, relpath)
+    release_request = factories.create_request_at_status(
+        "workspace",
+        author=author,
+        status=RequestStatus.RETURNED,
+    )
+
+    with pytest.raises(bll.RequestPermissionDenied, match=r"not updated"):
+        bll.update_file_in_request(release_request, relpath, author)
+
+
+@pytest.mark.parametrize(
+    "status,reviewed,approved,success,notification_sent",
+    [
+        (RequestStatus.PENDING, False, None, True, False),
+        (RequestStatus.SUBMITTED, False, None, False, False),
```

Comment:
`approved=None` doesn't mean anything different to `approved=False`; is it intended to?

Reply:
no, I just used that to differentiate cases where it wouldn't be used at all from cases where it would be set but be false. but actually with the tiny refactor from your other comment that becomes a moot point - I'll just set them to `False`.

Reply:
done